providers/google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/domain/Network.java

```diff
@@ -38,24 +46,39 @@
 
    public abstract String name();
 
+   public abstract NetworkType type();
+
    @Nullable public abstract String description();
 
    /**
     * The range of internal addresses that are legal on this network. This range is a CIDR
     * specification, for example: {@code 192.168.0.0/16}.
     */
-   public abstract String rangeIPv4();
+   @Nullable public abstract String rangeIPv4();
 
    /**
     * This must be within the range specified by IPv4Range, and is typically the first usable address in that range.
     * If not specified, the default value is the first usable address in IPv4Range.
     */
    @Nullable public abstract String gatewayIPv4();
 
-   @SerializedNames({ "id", "creationTimestamp", "selfLink", "name", "description", "IPv4Range", "gatewayIPv4" })
+   @Nullable public abstract List<URI> subnetworks();
+
+   public static Network create(String id, Date creationTimestamp, URI selfLink, String name, String description, String rangeIPv4,
+                                String gatewayIPv4) {
+      return new AutoValue_Network(id, creationTimestamp, selfLink, name, NetworkType.LegacyNetwork, description,
+              rangeIPv4, gatewayIPv4, null);
+   }
```

Comment:
This will be merged and released to 2.0, so we can expect some breaking changes. Trivial ones such as this one shouldn't worry us.
An option to make this more update-friendly for future releases would be to provide a builder with AutoValue.Builder. In any case, I think we can just break this in 2.0 and keep jsut one factory method (plus the builder if you want). Same applies for other similar comments on this.